js/apps/admin-ui/src/components/dynamic/UserProfileAttributeListComponent.tsx

```diff
@@ -0,0 +1,83 @@
+import type { UserProfileConfig } from "@keycloak/keycloak-admin-client/lib/defs/userProfileMetadata";
+import { FormGroup } from "@patternfly/react-core";
+import { useState } from "react";
+import { Controller, useFormContext } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { HelpItem } from "ui-shared";
+import { convertToName } from "./DynamicComponents";
+import { adminClient } from "../../admin-client";
+import { useFetch } from "../../utils/useFetch";
+import type { ComponentProps } from "./components";
+import { KeySelect, FetchCallback } from "../key-value-form/KeySelect";
+
+type UserProfileAttributeListProps = ComponentProps & {};
+
+export const UserProfileAttributeListComponent = ({
+  name,
+  label,
+  helpText,
+  defaultValue,
+  required = false,
+}: UserProfileAttributeListProps) => {
+  const { t } = useTranslation();
+  const {
+    control,
+    formState: { errors },
+  } = useFormContext();
+
+  const [config, setConfig] = useState<UserProfileConfig>();
+  const convertedName: string = convertToName(name!);
+
+  const [custom, setCustom] = useState(true);
+  const fetchCallback: FetchCallback = {
+    fieldValue: "",
+    custom: custom,
+    setCustom: setCustom,
+  };
+
+  useFetch(
+    () => adminClient.users.getProfile(),
+    (cfg) => {
+      setConfig(cfg);
+
+      const fieldIncluded = cfg.attributes
+        ?.map((option) => option.name!)
+        .includes(fetchCallback.fieldValue);
+      setCustom(!fieldIncluded);
+    },
+    [],
+  );
+
+  const convert = (config: UserProfileConfig | undefined) => [
+    ...(config?.attributes?.map((option) => ({
+      key: option.name!,
+      label: option.name!,
+    })) || []),
```

Comment:
I don't think we need to spread this array as `.map()` already returns a new array. Perhaps we could also make this slightly more readable with an early return:

```ts
const convert = (config?: UserProfileConfig) => {
  if (!config?.attributes) return [];

  return config.attributes.map((option) => ({
    key: option.name!,
    label: option.name!,
  }));
};
```

Reply:
I've updated this part in the latest version of the PR.